meta.yaml

```diff
@@ -0,0 +1,17 @@
+package:
+  name: latent-space-builder
+  version: 0.0.1
+
+source:
+  path: .
+
+requirements:
+  build:
+    - python
+    - setuptools
+
+  run:
+    - python
+
+about:
+  home:
```

Comment:
Is it something that we need?

Reply:
I followed the instructions from this link [https://docs.conda.io/projects/conda-build/en/latest/user-guide/tutorials/build-pkgs.html](https://docs.conda.io/projects/conda-build/en/latest/user-guide/tutorials/build-pkgs.html) and added this file in order to build a Conda package. 

When `conda build .` is executed from the Terminal, the package is uploaded and/or updated on Anaconda Cloud. The package then becomes available on [https://anaconda.org/rdeeban/latent-space-builder](https://anaconda.org/rdeeban/latent-space-builder). 

One benefit of uploading the package to Anaconda Cloud is that the package can be used to build the compSPI ```environment.yml```:

```
name: compSPI
channels:
  - conda-forge
  - omnia
  - defaults
  - rdeeban
dependencies:
  - python=3.7
  - jupyterlab
  - latent-space-builder
  - latent-space-visualizer
  - pip
  - pip:
    - numpy==1.19.0
    - pyDiffMap
    - kora
```
I tested the above ```environment.yml``` by creating an environment from the file. I then imported the `latent_space_builder` and `latent_space_visualizer` and ran the code that was from the notebook from the Terminal. 

However, I am unable to upload the `environment.yml` file to the `compSPI` organization on Anaconda Cloud.

I added the following line because I was receiving an unsatisfiable dependency error when this dependency was added to this file:
```
numpy==1.19.0
```

I added the following lines because these dependencies could not be installed under `requirements-run` section of this file:
```
pyDiffMap
kora
```

The reason I added support for building a Conda package is that when I tried to update the existing package on PyPI for this project ([https://packaging.python.org/tutorials/packaging-projects/](https://packaging.python.org/tutorials/packaging-projects/)), I received the following error:

```
(base) deebanramalingam@Deebans-MBP LatentSpaceBuilder % python -m twine upload dist/* --verbose
Uploading distributions to https://upload.pypi.org/legacy/
  dist/latent_space_builder-0.0.1-py3-none-any.whl (2.9 KB)
  dist/latent-space-builder-0.0.1.tar.gz (2.5 KB)
  dist/latent_space_builder-0.0.1-py3.7.egg (4.1 KB)
Uploading latent_space_builder-0.0.1-py3-none-any.whl
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8.02k/8.02k [00:00<00:00, 8.65kB/s]
Content received from server:
<html>
 <head>
  <title>400 File already exists. See https://pypi.org/help/#file-name-reuse for more information.</title>
 </head>
 <body>
  <h1>400 File already exists. See https://pypi.org/help/#file-name-reuse for more information.</h1>
  The server could not comply with the request since it is either malformed or otherwise incorrect.<br/><br/>
File already exists. See https://pypi.org/help/#file-name-reuse for more information.


 </body>
</html>
HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/
File already exists. See https://pypi.org/help/#file-name-reuse for more information.
```

Reply:
OK, I like the idea but this sounds complicated at the moment, let's discuss